ocaml/xapi/create_misc.ml

```diff
@@ -46,6 +45,49 @@ type host_info = {
 	ssl_legacy: bool;
 }
 
+(** The format of the response looks like
+ *  # xen-livepatch list
+ *   ID                                     | status
+ *  ----------------------------------------+------------
+ *  hp_1_1                                  | CHECKED
+ *  hp_2_1                                  | APPLIED
+ *  hp_3_2                                  | APPLIED *)
+let make_xen_livepatch_list () = 
+	let lines = try Xstringext.String.split '\n' (Helpers.get_process_output !Xapi_globs.xen_livepatch_list) with _ -> [] in
+ 	let patches = List.fold_left(
+		fun acc l ->
+		match List.map String.trim (Xstringext.String.split ~limit:2 '|' l) with
+		| [ key; "APPLIED" ] -> key :: acc
+		| _ -> acc;
+	)[] lines in
+	if List.length patches > 0 then Some(String.concat ", " patches) else None
+
+(** The format of the response looks like
+ *  # kpatch list
+ *  Loaded patch modules:
+ *  kpatch_hp_1_1
+ *  kpatch_hp_2_1
+
+ *  Installed patch modules: *)
+let make_kpatch_list () = 
+	let start_line = "Loaded patch modules:" in 
+	let end_line = "Installed patch modules:" in
+	let lines = try Xstringext.String.split '\n' (Helpers.get_process_output !Xapi_globs.kpatch_list) with _ -> []in
+	let rec loop acc started = function
+  		| []                                  -> acc
+  		| line :: _    when line = end_line   -> acc
+  		| line :: rest when line = start_line -> loop acc true rest
+  		| line :: rest ->
+    	let line' = String.trim line in
```

Comment:
The indentation is wrong in this function. It helps to use just tabs at the beginning of lines.